lib/radiator_web/controllers/login_controller.ex

```diff
@@ -0,0 +1,137 @@
+defmodule RadiatorWeb.LoginController do
```

Comment:
It feels to me like this controller has to many responsibilities. Also, if I can, I try to stay to the "standard" controller actions.

An abstraction I found useful is to model login around "sessions" as this is an object that corresponds well with the standard controller actions: new (form), create (login/auth action), delete (logout).

Signup should be in a separate RegistrationController. Even the verification flow might benefit from being extracted into its own controller.

Reply:
Okay, in that model though if I would want a combined login/signup page, who would be the server of that?

Not sure if I like the idea of the session being a resource style thing, as it is fleeting and not permanent, which would be the separation in my head. Also isn't that misleading? We do have a session for each contact, regardless if they are logged in or not, don't we?

Reply:
Hm true, technically the session is there even for logged out users, the login action only _alters_ the session.. Then ... `AuthenticationController` maybe? But then you could also leave it as `LoginController`.

Regarding combined pages: Not sure, possibly another controller just for rendering the combined form/page? The actual login/register actions I would still keep in separate controllers.